crates/ruff/src/rules/pylint/rules/redefined_loop_name.rs

```diff
@@ -0,0 +1,157 @@
+use crate::ast::helpers::find_names;
+use crate::ast::types::{Node, Range};
+use crate::ast::visitor;
+use crate::ast::visitor::Visitor;
+use crate::checkers::ast::Checker;
+use crate::registry::Diagnostic;
+use crate::source_code::Locator;
+use crate::violation::Violation;
+use ruff_macros::{define_violation, derive_message_formats};
+use rustpython_parser::ast::{Expr, Stmt, StmtKind, Withitem};
+use std::iter::zip;
+
+define_violation!(
+    pub struct RedefinedLoopName {
```

Comment:
Please add documentation here (so that the rule docs can be autogenerated). For an example, look at [yield-in-init (PLE0100)](https://beta.ruff.rs/docs/rules/yield-in-init/)